shinylive/_deps.py

```diff
@@ -397,45 +408,77 @@ def _find_recursive_deps(
     packages passed in.
     """
     pyodide_lock = _pyodide_lock_data()
-    deps = list(pkgs)
+
+    # The keys in pyodide_lock are not the same as the package names. For example, the
+    # key "jsonschema-specifications" points to an object where the "name" entry is
+    # "jsonschema_specifications". The dependencies are listed with names, not keys.
+
+    dep_names = list(dep_names)
     i = 0
-    while i < len(deps):
-        dep = deps[i]
-        if dep not in pyodide_lock["packages"]:
-            # TODO: Need to distinguish between built-in packages and external ones in
-            # requirements.txt.
-            verbose_print(
-                f"  {dep} not in pyodide-lock.json. Assuming it is in base Pyodide or in requirements.txt."
-            )
-            deps.remove(dep)
+    while i < len(dep_names):
+        dep_name = dep_names[i]
+        dep_key: str | None = dep_name_to_dep_key(dep_name)
+
+        if dep_key not in pyodide_lock["packages"]:
+            if dep_name not in BASE_PYODIDE_PACKAGE_NAMES:
+                # TODO: Need to distinguish between built-in packages and external ones in
+                # requirements.txt.
+                verbose_print(
+                    f"  {dep_name} not in pyodide-lock.json. Assuming it is in base Pyodide or in requirements.txt."
+                )
+            dep_names.remove(dep_name)
             continue
 
-        dep_deps = set(pyodide_lock["packages"][dep]["depends"])
-        new_deps = dep_deps.difference(deps)
-        deps.extend(new_deps)
+        dep_depnames = set(pyodide_lock["packages"][dep_key]["depends"])
+        new_depnames = dep_depnames.difference(dep_names)
+        dep_names.extend(new_depnames)
         i += 1
 
-    return deps
+    return dep_names
 
 
-def _dep_name_to_dep_file(dep_name: str) -> str:
+def dep_name_to_dep_key(name: str) -> str | None:
     """
-    Given the name of a dependency, like "pandas", return the name of the .whl file,
-    like "pandas-1.4.2-cp310-cp310-emscripten_3_1_14_wasm32.whl".
+    Convert a package name to a key that can be used to look up the package in
+    pyodide-lock.json.
+
+    The keys in pyodide-lock.json are not the same as the package names. For example,
+    the key "jsonschema-specifications" points to an object where the "name" entry is
+    "jsonschema_specifications".
+
+    Note that the names are lowercased because the package names should be treated as
+    case-insensitive. https://github.com/pyodide/pyodide/issues/1614
     """
-    pyodide_lock = _pyodide_lock_data()
-    return pyodide_lock["packages"][dep_name]["file_name"]
+    # Special case for base pyodide packages
+    if name in BASE_PYODIDE_PACKAGE_NAMES:
+        return name
+
+    name = name.lower()
+    if name not in _dep_name_to_dep_key_mappings():
+        return None
```

Comment:
Should this maybe throw a warning?

Reply:
Oh nvm, now I see the warning above

Reply:
It will be fairly common to not have a package in the pyodide_lock.json file. In that case, micropip will go out and try to install it from PyPI.